include/mxnet/tuple.h

```diff
@@ -106,7 +113,10 @@ class Tuple {
   inline void assign(RandomAccessIterator begin,
                      RandomAccessIterator end) {
     this->SetDim(end - begin);
-    std::copy(begin, end, this->begin());
+    CHECK_GE(ndim(), 0);
+    if (ndim() > 0) {
```

Comment:
Don't need check `ndim() > 0` actually

Reply:
Good catch. I was fixing another assignment problem and just added the check everywhere. Will remove this.